cpp/tests/CMakeLists.txt

```diff
@@ -1,19 +1,19 @@
 if(logging)
   add_library(common_catch_main_object OBJECT "common_catch_main.cc")
-  if(SPDLOG_INCLUDE_DIR)
-    target_include_directories(common_catch_main_object SYSTEM PUBLIC ${SPDLOG_INCLUDE_DIR})
+  if(spdlog_INCLUDE_DIR)
```

Comment:
This seems like the wrong way to check if we need spdlog, if that's what it's doing. Shouldn't this be `spdlog_FOUND` or `logging`?

Similar comments to `CATCH` and `sopt` below

Reply:
I don't really understand the logic of having these if statements, but I'm going to leave them alone for now. If the include dirs aren't found they don't get included in the target, but surely the build will then fail? Is it possible that they have been provided through some other way?